### ISSUE TITLE
Version tags

### ISSUE THREAD

**Issue by CelestialWalrus** [karlstav/cava#17]
We should tag stable versions (i.e. for the purposes of packaging).

I've made an AUR package for git version: https://aur.archlinux.org/packages/cava-git and it'd be good to make a stable package.

**Comment by anko** [comment]
:+1:

[semver](http://semver.org/)?

**Comment by CelestialWalrus** [comment]
We should start with 0.1.0, I'll try to get cava into community repo.

**Comment by karlstav** [comment]
OK so semver it is then? It think it seems intuitive and reasonable, but I can see that there is some [resistance](https://gist.github.com/jashkenas/cbd2b088e20279ae2c8e).

Could someone explain exactly how it should be implemented? I have already merged @livibetter's #18 which prints the version number from the tag on github. I guess now it only uses the tag on the latest commit (which is now '04f0d4d'). 

So should this tag be changed, by me or someone else, according to the specifications of semver every time something is merged in the master branch?

**Comment by anko** [comment]
> there is some [resistance](https://gist.github.com/jashkenas/cbd2b088e20279ae2c8e).

The "resistance" is mainly just jashkenas. I think his arguments are weak. [Related HN thread.](https://news.ycombinator.com/item?id=8244700)

> We should start with 0.1.0

@CelestialWalrus Or 0.0.1, or 0.0.0. But whatever. :) The main point is that semver 0.x.x means "not guaranteeing any API yet", which makes sense for figuring out how things will work.

Speaking of which:

> Could someone explain how it should be implemented?

A common arrangement is that the owner or other project maintainer (that would be you @karlstav) increments the version appropriately before making a release.

As for "making a release": Github has a built-in thing that if you push [git tags](http://git-scm.com/book/en/v2/Git-Basics-Tagging) that look like semver, they'll be added under [the Releases tab](https://github.com/karlstav/cava/releases) on the repository's page, which [looks like this when populated](https://github.com/anko/whatxml/releases). There's even [a convenient API](https://developer.github.com/v3/repos/releases/) for fetching them, which can help with packaging.

The current makefile (which runs `git describe` and plugs it into what's output by `-v`) should work very well, since `git describe` picks up the latest tags in a semver-compatible way. (If I do `git tag v0.1.0`, followed by `make clean; make; ./cava -v`, it prints `cava v1.0.1`.)

In summary, all that would be needed is to add an appropriate git tag and push it. Everything else "magically" works.

**Comment by livibetter** [comment]
I didn't read that post, it's too long.

I think as long as you stick to something you decide, then it's fine. So, if we go with semver, then we go with semver 100%. I have been using it for long time, never have problems when tagging a new release. simple rules, only three questions I have to ask myself: Is it bugfix or functionality unchanged? Is it adding new stuff? Is it breaking something?

Life is much easier with semver.

----

First version, well, you can go with 2014.5.4 (codename: MAYTHEFOURTHBEWITHYOU) if you really want. That really doesn't matter as long as you stick to semver for next version.

But 0.1.0, 0.0.1, or 0.0.0 make senses for first version.

----

@anko has explained it well, you just `git tag` whenever a new release are ready, then push out, and you are done.

By the way, if you don't like the look of `cava v3.1.4`, I can edit it. I personally feel weird to have "v" even myself always tag with "v" prefix for version tags.

We can also add -- for example `ffempg -version`:

* `make` (or build) date/time and/or GCC and/or system info
* compilation/configuration options (currently only `debug`)

**Comment by CelestialWalrus** [comment]
@karlstav you should tag the first version soon. I'll try to get it into official repos. (Arch & others)

**Comment by karlstav** [comment]
It's officially now 0.1.0! Or should it be v0.1.0? Anyway, great work everyone, thanks for all help so far making cava more official.

**Comment by CelestialWalrus** [comment]
@karlstav thanks, I'll make a stable AUR package.

**Comment by karlstav** [comment]
seems to be a matter of opinion so we will go for no "v", people should now what these three dot-separated numbers mean by now.

**Comment by karlstav** [comment]
closing this as we are now using version tags.
